x/mint/types/inflations_test.go

```diff
@@ -0,0 +1,181 @@
+package types_test
+
+import (
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"math"
+	"math/rand"
+	"strconv"
+	"testing"
+)
+
+func resetSupply(app *simapp.SimApp, ctx sdk.Context, initSupply sdk.Coins, curSupply sdk.Coins) {
+	err := app.BankKeeper.MintCoins(ctx, types.ModuleName, initSupply)
+	if err != nil {
+		panic(err)
+	}
+	err = app.BankKeeper.BurnCoins(ctx, types.ModuleName, curSupply)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getTestingAccounts(r *rand.Rand, n int, ctx sdk.Context, app *simapp.SimApp) []simtypes.Account {
+	accounts := simtypes.RandomAccounts(r, n)
+
+	for _, account := range accounts {
+		acc := app.AccountKeeper.NewAccountWithAddress(ctx, account.Address)
+		app.AccountKeeper.SetAccount(ctx, acc)
+	}
+
+	return accounts
+}
+
+func TestHandleInflations(t *testing.T) {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	acc := auth.NewEmptyModuleAccount(types.ModuleName, auth.Minter, auth.Burner)
+	app.AccountKeeper.SetModuleAccount(ctx, acc)
+
+	s := rand.NewSource(1)
+	r := rand.New(s)
+	targetAccounts := getTestingAccounts(r, 3, ctx, app)
+
+	minter := types.DefaultInitialMinter()
+	params := types.DefaultParams()
+
+	app.MintKeeper.SetParams(ctx, params)
+	app.MintKeeper.SetMinter(ctx, minter)
+
+	testDenom := "testDenom"
+	initSupplyAmount, _ := sdk.NewIntFromString("1000000000000000000")
```

Comment:
It would be better, if the amount of token supply used for testing would be in the same orddr of magnitude as real tokens - foe example, we have cc. `10**9 [FET] = 10**27 [aFET] = 1000000000000000000000000000 [aFET]`.

This is just to make sure that computational errors we are getting during the test are caused by amount tokens in the same order of magnitude as in reality.

This would require to change values of expected minted tokens a few lines below.